src/App.tsx

```diff
@@ -83,9 +83,11 @@ function App() {
 
 function AppWrapper() {
   return (
-    <Router>
-      <App />
-    </Router>
+    <CurrencyProvider>
+      <Router>
+        <App />
+      </Router>
+    </CurrencyProvider>
```

Comment:
this feels a bit odd to me, if at some point we wanted another global provider/context how hard is it for us to add it

Reply:
You can just keep wrapping it in contexts

Reply:
I feel like that could get ugly later but we should deal with that if it becomes an issue